pytensor/link/pytorch/dispatch/blockwise.py

```diff
@@ -0,0 +1,28 @@
+import torch
+
+from pytensor.graph import FunctionGraph
+from pytensor.link.pytorch.dispatch import pytorch_funcify
+from pytensor.tensor.blockwise import Blockwise
+
+
+@pytorch_funcify.register(Blockwise)
+def funcify_Blockwise(op: Blockwise, node, *args, **kwargs):
+    batched_dims = op.batch_ndim(node)
+    core_node = op._create_dummy_core_node(node.inputs)
+    core_fgraph = FunctionGraph(inputs=core_node.inputs, outputs=core_node.outputs)
+    core_func = pytorch_funcify(core_fgraph)
+    if len(node.outputs) == 1:
+
+        def inner_func(*inputs):
+            return core_func(*inputs)[0]
+    else:
+        inner_func = core_func
+
+    for _ in range(batched_dims):
+        inner_func = torch.vmap(inner_func)
```

Comment:
Does this handle broadcasting of the batch dimensions?

Say you have a @ b with shapes `(5, 1, 2, 3)` and `(1, 7, 3, 4)`, where output should be `(5, 7, 3, 4)`

Reply:
It does, hopefully the test below helps